### PR TITLE
Change thrift for PollForActivityTask to include throttling limit

### PR DESCRIPTION
**Pull request by madhuravi**
This only includes thrift changes in case there are questions on that.
I did not change the "TaskList" structure directly because it is used in a number of other APIs, and it would be more confusing and error prone to have a throttling RPS on all these APIs (eg. StartChildWorkflow, ScheduleActivity etc.)

### CODE REVIEW

File: idl/github.com/uber/cadence/shared.thrift [hunk 1]
```diff
@@ -177,6 +177,10 @@ struct TaskList {
   10: optional string name
 }
 
+struct TaskListMetadata {
+  10: optional i64 maxTasksPerSecond
```
Comment (mfateev) [inline idl/github.com/uber/cadence/shared.thrift]: It should be non integer. For example it should be possible to specify 0.1 per second to get one in 10 seconds.

File: idl/github.com/uber/cadence/shared.thrift [hunk 2]
```diff
@@ -713,6 +717,7 @@ struct PollForActivityTaskRequest {
   10: optional string domain
   20: optional TaskList taskList
   30: optional string identity
+  40: optional TaskListMetadata taskListMetadata
```
Comment (mfateev) [inline idl/github.com/uber/cadence/shared.thrift]: I'm confused how does task list relate to a workflow execution? Multiple workflow types can share a single task list. The same for activity task list. Last time we talked the task list throttling metadata belonged to Domain configuration.

Comment (samarabbas) [reply idl/github.com/uber/cadence/shared.thrift]: @mfateev this is about rate limiting the tasks dispatched from tasklist.  If user wants to rate limit individual activity type they can create separate tasklist for them.

Comment (mfateev) [reply idl/github.com/uber/cadence/shared.thrift]: Code reviewer made it look this field belonged to WorkflowExecution structure. I still don't believe that this belongs to Poll request as it is not clear what semantic would be of multiple workers passing different values. Also we don't want to redeploy workers to change the rate. IMHO it should be part of Domain metadata.

Comment (madhuravi) [reply idl/github.com/uber/cadence/shared.thrift]: Not sure what you mean by WorkflowExecution structure. I agree that it doesn't belong in the Poll request but this was the conclusion from the meeting we had yesterday. I have shared the doc with you, I can send it again as well. Do you mind taking a look and adding your comments there?

**Comment by madhuravi** [comment]
@samarabbas @mfateev Looks good post-meeting?

File: idl/github.com/uber/cadence/shared.thrift [hunk 3]
```diff
@@ -177,6 +177,11 @@ struct TaskList {
   10: optional string name
 }
 
+struct TaskListMetadata {
+  10: optional double maxTasksPerSecond
+  20: optional i64 version
```
Comment (mfateev) [inline idl/github.com/uber/cadence/shared.thrift]: Add the comment explaining the version behavior.

**Pull request merged**